### PR TITLE
Upload user photo

### PR DESCRIPTION
**Pull request by AlexanderKhapchenko**

### CODE REVIEW

File: backend/src/middlewares/photo/photo.middleware.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      const fileSize = Math.round(file.size / 1024);
+
+      if (!validation.includes(file.type)) {
+        throw new Error('This is not an Image File!');
+      } else if (fileSize > 10240) {
+        throw new Error('File too Big, please select a file less than 10mb');
+      }
+
+      next();
+    }
```
Comment (o-rudych) [inline backend/src/middlewares/photo/photo.middleware.ts]: Too many nested `if` blocks. You can do it like:
```
if (!req.files) {
  return next(); // or what you need to return
}
const file = req.files.photo;
...
```

File: frontend/src/components/edit-profile/edit-profile.tsx [hunk 2]
```diff
@@ -102,6 +119,25 @@ export const EditProfile: React.FC<EditProfileProps> = ({ onClose }) => {
...
+    if (!fileList) return;
+
+    const file = fileList[0];
+    const validation = ['image/jpeg', 'image/jpg', 'image/png'];
+    const fileSize = Math.round(file.size / 1024);
+    if (!validation.includes(file.type)) {
+      alert('This is not an Image File!');
+    } else if (fileSize > 10240) {
+      alert('File too Big, please select a file less than 10mb');
+    }
```
Comment (o-rudych) [inline frontend/src/components/edit-profile/edit-profile.tsx]: Maybe it's better to move the validation rules to shared folder?
And use them both on the backend and on the frontend.

**Review by o-rudych — CHANGES_REQUESTED** [review]

File: backend/src/controllers/update-user/update-user.controller.ts [hunk 3]
```diff
@@ -84,4 +85,29 @@ const deleteOauthConnections = async (
...
+): Promise<Response | undefined> => {
+  try {
+    if (req.files) {
+      const photoUrl = await uploadFileToS3(req.files.photo);
+      await userService.updateUserPhoto(req.body.tokenPayload.sub, photoUrl);
+      return res.status(httpStatus.OK).json({ photoUrl });
+    }
+
+    throw new Error('Failed upload S3');
+  } catch (error) {
```
Comment (oleksandr-danylchenko) [inline backend/src/controllers/update-user/update-user.controller.ts]: Please use guard clause:
```ts
if (!req.files) throw new Error('Failed upload S3');

...rest of the code...
```

File: backend/src/middlewares/photo/photo.middleware.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { ErrorMessage } from '@autoline/shared';
+import { NextFunction, Response } from 'express';
+import httpStatus from 'http-status-codes';
+
+const photoMiddleware = async (
+  /* eslint-disable */
```
Comment (oleksandr-danylchenko) [inline backend/src/middlewares/photo/photo.middleware.ts]: No! Disable eslint only of the next line, not the whole file!

File: backend/src/middlewares/photo/photo.middleware.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+      if (!validation.includes(file.type)) {
+        throw new Error('This is not an Image File!');
+      } else if (fileSize > 10240) {
+        throw new Error('File too Big, please select a file less than 10mb');
+      }
+
+      next();
+    }
+  } catch (err) {
```
Comment (oleksandr-danylchenko) [inline backend/src/middlewares/photo/photo.middleware.ts]: Guard clauses

File: frontend/src/components/edit-profile/edit-profile.tsx [hunk 6]
```diff
@@ -89,6 +98,15 @@ export const EditProfile: React.FC<EditProfileProps> = ({ onClose }) => {
...
   };
 
+  const submitAvatar = async (): Promise<void> => {
+    if (selectedAvatar) {
+      const formData = new FormData();
+      formData.append('photo', selectedAvatar, selectedAvatar.name);
+      await updateUserPhoto(formData);
+      setPreviewAvatar('');
+    }
+  };
```
Comment (oleksandr-danylchenko) [inline frontend/src/components/edit-profile/edit-profile.tsx]: Guard clauses

File: frontend/src/components/edit-profile/edit-profile.tsx [hunk 7]
```diff
@@ -102,6 +120,25 @@ export const EditProfile: React.FC<EditProfileProps> = ({ onClose }) => {
...
+    if (!fileList) return;
+
+    const file = fileList[0];
+    const validation = ['image/jpeg', 'image/jpg', 'image/png'];
+    const fileSize = Math.round(file.size / 1024);
+    if (!validation.includes(file.type)) {
+      alert('This is not an Image File!');
+    } else if (fileSize > 10240) {
+      alert('File too Big, please select a file less than 10mb');
+    }
```
Comment (oleksandr-danylchenko) [inline frontend/src/components/edit-profile/edit-profile.tsx]: Why don't you stop the execution of the function when the error condition is met? `alert` just shows the message, but below you still create the object URL and set it as avatar

**Review by oleksandr-danylchenko — CHANGES_REQUESTED** [review]

**Review by oleksandr-danylchenko — APPROVED** [review]

**Pull request merged**